sdk/cosmosdb/cosmos/src/request/DedicatedGatewayRequestOptions.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Dedicated Gateway Request Options
+ */
+export interface DedicatedGatewayRequestOptions {
```

Comment:
[Blocking]
What's the beta equivalent of Node? 
This feature is still in beta, both .NET and JAVA are shipping with beta versions (Preview, @Beta).

Reply:
We don't really have a story for "beta APIs" - we only have beta versions of packages. So when we have a new major/minor that is not finalized, we will ship `beta.1` ... `beta.n` until it stabilizes and then GA .

Reply:
I will follow up offline.

Reply:
@xirzec - this approach won't work for Cosmos DB SDK. Please see our other SDKs as well (Java, python) - which also sit in azure sdk mono repo. 
Reason this won't work for us is because this `DedicatedGateway` feature can be in beta phase for 1 or 2 years. We cannot keep shipping beta packages for so long, as customers don't take beta packages in production, and also, we don't fully support beta packages for production use cases. 

We need something which is explicit for a feature marked as Beta, while the whole library gets shipped as GA library. 
Please see this reference issue - https://github.com/Azure/azure-sdk/issues/2593

Reply:
@xirzec - this approach won't work for Cosmos DB SDK. Please see our other SDKs as well (Java, python) - which also sit in azure sdk mono repo. 
Reason this won't work for us is because this `DedicatedGateway` feature can be in beta phase for 1 or 2 years. We cannot keep shipping beta packages for so long, as customers don't take beta packages in production, and also, we don't fully support beta packages for production use cases. 

We need something which is explicit for a feature marked as Beta, while the whole library gets shipped as GA library. 
Please see this reference issue - https://github.com/Azure/azure-sdk/issues/2593

Reply:
@kushagraThapar understood. In consideration of your circumstances, I propose the following policy:

1. We release this feature inside a GA package version (no beta suffix)
2. The feature is marked as "experimental" (or beta if you prefer) through reference documentation / comments
3. If you decide to make a breaking change to an experimental feature, we will require a major version bump of the package in order to adhere to our semantic versioning policies and allow customers a chance to fix their code.

Are these terms acceptable?

Reply:
@xirzec  - first 2 policies make sense to me, however I don't quite agree with 3rd policy. 
Thinking about a scenario where we will need to change the API 2-3 times (which has happened in the past in Java SDK at least), that would result in 2-3 new Cosmos Db Node JS SDK versions, which will then create so much chaos. 

One hybrid approach that we settled on with Azure arch board regarding beta APIs was that if there are breaking changes in the Beta APIs, we introduce new APIs, and we mark the existing APIs deprecated, keep them in next 2 minor versions, giving customers a chance to fix their code and upgrade. 
Would this work on NodeJS SDK?

Reply:
The guarantee we are trying to give our customers is that if they depend on an @azure package (e.g. `"@azure/cosmos": "^3.15.1"` inside their package.json dependencies) we will never break a TS build without doing a major version bump. There are some cases where we change behavior significantly enough that we do a minor version bump, but we'd still want customers who use `~3.15.1` as their version specifier to get patch level bugfixes without worry.

I know other projects (notably TypeScript itself) are a bit more lax around what is a minor versus major change, but really we don't want folks to have to start pinning versions or individually vetting each release as this will delay critical bugfixes and security fixes from being readily adopted.

@bterlson do you have any suggestions about how Cosmos might mitigate risk here?

Reply:
Please follow up with me offline. Removing the blocker. I add a comment for experimental.

Reply:
Alright I am back with new guidance that I think will work for your scenario. Basically, we'll be aligning with how Office handles this:

1. Experimental APIs must be tagged with `@beta` so api-extractor can identify them
2. After experimental APIs have been marked as `@deprecated` for two releases they may be removed *without* a major bump, but there *will* be a minor version bump.

I believe the second point aligns with @kushagraThapar 's proposal above? @jay-most how do you feel about this?

Reply:
@xirzec - this is great, and aligns with our model of Beta APIs in Java SDK as well! Thank you for the guidance. 
@jay-most  - can you please make this change?